appservice/src/createAppService/SiteNameStep.ts

```diff
@@ -48,12 +61,24 @@ export class SiteNameStep extends AzureNameStep<IAppServiceWizardContext> {
             prompt = vscode.l10n.t('Enter a globally unique name for the new web app.');
         }
 
-        context.newSiteName = (await context.ui.showInputBox({
+        const options: AgentInputBoxOptions = {
             prompt,
             placeHolder,
             validateInput: (name: string): string | undefined => this.validateSiteName(name),
-            asyncValidationTask: async (name: string): Promise<string | undefined> => await this.asyncValidateSiteName(client, name)
-        })).trim();
+            asyncValidationTask: async (name: string): Promise<string | undefined> => await this.asyncValidateSiteName(client, name),
+            agentMetadata: this._inputBoxOptionsAgentMetadata ?? {
+                parameterName: "siteName",
+                parameterDisplayTitle: "Site Name",
+                parameterDisplayDescription: "The name of the app service site.",
+            }
+        };
+
+        if (this._skipValidation) {
```

Comment:
I think that this might be a pain to go in and implement to other name steps later.

We don't need to do this now, but I was thinking that we could give `AzureNameStep` a `formatOptions` function that would just do this for you. Maybe something like this:

```

const options: AgentInputBoxOptions = this.formatOptions({
            prompt,
            placeHolder,
            validateInput: (name: string): string | undefined => this.validateSiteName(name),
            asyncValidationTask: async (name: string): Promise<string | undefined> => await this.asyncValidateSiteName(client, name),
            agentMetadata: this._inputBoxOptionsAgentMetadata ?? {
                parameterName: "siteName",
                parameterDisplayTitle: "Site Name",
                parameterDisplayDescription: "The name of the app service site.",
            }
        });

.......


protected formatOptions(options: AzExtInputBoxOptions ): AzExtInputBoxOptions {
  if (this._skipValidation) {
    options.validateInput = undefined;
    options.asyncValidationTask = undefined;
  }
}
```

I could see in the future that we may want to have other options that automatically get formatted/adjusted, so just keeping the name generic.

Reply:
I think I will dump this part of the change for now. Looking at how validation is handled, it is handled by `AzExtUserInput`, which is out of the question.